smartsim/_core/entrypoints/dragon.py

```diff
@@ -90,39 +95,63 @@ def print_summary(network_interface: str, ip_address: str) -> None:
                 HOSTNAME: {socket.gethostname()}
                 DRAGON_SERVER_CONFIG: {json.dumps(zmq_config)}
                 --------------------------------------
+                """),
+        )
 
-                --------------- Output ---------------
 
-                """),
+def restart_updater(
+    backend: DragonBackend, updater: t.Optional[ContextThread]
+) -> ContextThread:
+    # If the updater was started, check if it completed or died
+    if updater is not None:
+        updater.join(0.1)
+        # If it's alive, there is nothing to do
+        if updater.is_alive():
+            return updater
```

Comment:
It seems wise to avoid ever returning the existing thread when requesting a restart (in case there is some saved state). This may be due to my interpretation of the method name and lack of docstring.